src/vm/hints/find_element_hint.rs

```diff
@@ -0,0 +1,558 @@
+            Ok(())
+        );
+
+        assert_eq!(
+            vm.memory.get(&MaybeRelocatable::from((0, 3))),
+            Ok(Some(&MaybeRelocatable::Int(bigint!(1))))
+        )
+    }
+
+    #[test]
+    fn element_not_found_search() {
+        let (mut vm, ids) = init_vm_ids();
+        vm.memory.data[0][4] = Some(MaybeRelocatable::from(bigint!(7)));
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::FindElemKeyNotFound(
+                MaybeRelocatable::Int(bigint!(7))
+            ))
+        );
+    }
+
+    #[test]
+    fn element_not_found_global() {
+        let (mut vm, ids) = init_vm_ids();
+        vm.find_element_index = Some(bigint!(2));
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::FindElemNoFoundKey)
+        );
+    }
+
+    #[test]
+    fn find_elm_incorrect_ids() {
+        let (mut vm, mut ids) = init_vm_ids();
+        ids.remove(&"array_ptr".to_string());
+
+        assert!(matches!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::IncorrectIds(_, _))
+        ));
+    }
+
+    #[test]
+    fn find_elm_failed_ids_get_addres() {
+        let (mut vm, ids) = init_vm_ids();
+        vm.references.insert(
+            0,
+            HintReference {
+                register: Register::FP,
+                offset1: -7,
+                offset2: 0,
+                inner_dereference: false,
+                ap_tracking_data: None,
+            },
+        );
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::FailedToGetIds)
+        );
+    }
+
+    #[test]
+    fn find_elm_failed_ids_get_from_mem() {
+        let (mut vm, ids) = init_vm_ids();
+        vm.memory.data[0][2] = None;
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::FailedToGetIds)
+        );
+    }
+
+    #[test]
+    fn find_elm_builtin_is_none() {
+        let (mut vm, ids) = init_vm_ids();
+        _ = vm.builtin_runners.pop();
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::NoRangeCheckBuiltin)
+        );
+    }
+
+    #[test]
+    fn find_elm_range_check_not_present() {
+        let (mut vm, ids) = init_vm_ids();
+        _ = vm.builtin_runners.pop();
+        vm.builtin_runners.push((
+            "output".to_string(),
+            Box::new(OutputBuiltinRunner::new(true)),
+        ));
+
+        assert_eq!(
+            execute_hint(&mut vm, FIND_ELEMENT_HINT, ids, &ApTracking::new()),
+            Err(VirtualMachineError::NoRangeCheckBuiltin)
+        );
+    }
```

Comment:
Add test with range check as second builtin. Should trigger the bug mentioned above.